src/util/numerics/mpz.h

```diff
@@ -33,6 +33,7 @@ class mpz {
     explicit mpz(int v) { mpz_init_set_si(m_val, v); }
     explicit mpz(uint64 v);
     explicit mpz(int64 v);
+    explicit mpz(double v) { mpz_init_set_d(m_val, v); }
```

Comment:
This seems to set an FPU error if called with `v = inf`. I think that's ok, since I think `x / 0` also sets an FPU error, and the contract of the API is to Not Do That.

Reply:
Yeah errors are ok here, it's not worth trying to make a error-safe api for floats

Reply:
Does the FPU error affect anything?